src/server/modules/threat/FileCheckController.ts

```diff
@@ -22,7 +22,33 @@ export class FileCheckController {
     this.virusScanService = virusScanService
   }
 
-  public checkFile: (
+  public fileExtensionCheck =
+    (allowedExtensions?: string[]) =>
+    async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+      const file = req.files?.file
+
+      if (Array.isArray(file)) {
```

Comment:
This logic seems to be duplicated multiple times, can we abstract this out?

Reply:
abstracted it out as another middleware